1. [Best practices for benchmarking germline small-variant calls in human genomes](https://www.nature.com/articles/s41587-019-0054-x)

2. [GA4GH Benchmarking Team GitHub repository](https://github.com/ga4gh/benchmarking-tools)

3. [hap.py benchmarking toolkit](https://github.com/Illumina/hap.py)

4. [Zhao S, Agafonov O, Azab A, et al. Accuracy and efficiency of germline variant calling pipelines for human genome data[J]. bioRxiv, 2020.](https://www.nature.com/articles/s41598-020-77218-4)

5. We can run [VCAT](https://www.illumina.com/products/by-type/informatics-products/basespace-sequence-hub/apps/variant-calling-assessment-tool.html) to produce recall and precision metrics for SNV and Indel.